go/adbc/ext.go

```diff
@@ -78,3 +81,55 @@ type OTelTracing interface {
 	// Gets the initial span attributes for any newly started span.
 	GetInitialSpanAttributes() []attribute.KeyValue
 }
+
+// IngestStreamOption bundles the most-common IngestStream settings.
+// Any other ADBC options can go into Extra.
+type IngestStreamOption struct {
+	TargetTable string            // required
+	IngestMode  string            // required, e.g. adbc.OptionValueIngestModeCreateAppend, or OptionValueIngestModeReplace
+	Extra       map[string]string // any other stmt.SetOption(...) args
```

Comment:
since these are required, should we make them explicit parameters instead of behind this struct? i.e.

```go
func IngestStream(ctx context.Context, cnxn Connection, reader array.RecordReader, target, mode string, extra map[string]string) (int64, error)
```

what do you think? @lidavidm thoughts?